.travis.yml

```diff
@@ -57,10 +58,28 @@ script:
   # Get AWS credentials when not a pull request build.
   - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then mkdir credentials; fi
   - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then wget https://www.amazontrust.com/repository/AmazonRootCA1.pem -O credentials/AmazonRootCA1.pem; fi
-  - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then echo -e $AWS_IOT_CLIENT_CERT > credentials/clientCert.pem; fi
-  - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then echo -e $AWS_IOT_PRIVATE_KEY > credentials/privateKey.pem; fi
+  - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
+      if [ "$RUN_TEST" == "provisioning" ]; then
+        echo -e $AWS_IOT_GAMMA_CLIENT_CERT > credentials/clientCert.pem;
+      else
+        echo -e $AWS_IOT_CLIENT_CERT > credentials/clientCert.pem;
+      fi
+    fi
+  - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then 
+      if [ "$RUN_TEST" == "provisioning" ]; then
+        echo -e $AWS_IOT_GAMMA_PRIVATE_KEY > credentials/privateKey.pem;
+      else
+        echo -e $AWS_IOT_PRIVATE_KEY > credentials/privateKey.pem;
+      fi
+    fi
   # Set credential defines passed to CMake when not a pull request build.
-  - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then export AWS_IOT_CREDENTIAL_DEFINES="-DIOT_TEST_SERVER=\"\\\"$AWS_IOT_ENDPOINT\\\"\" -DIOT_TEST_PORT=443 -DIOT_TEST_ROOT_CA=\"\\\"../credentials/AmazonRootCA1.pem\\\"\" -DIOT_TEST_CLIENT_CERT=\"\\\"../credentials/clientCert.pem\\\"\" -DIOT_TEST_PRIVATE_KEY=\"\\\"../credentials/privateKey.pem\\\"\""; fi
+  - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
+      if [ "$RUN_TEST" == "provisioning" ]; then 
+        export AWS_IOT_CREDENTIAL_DEFINES="-DIOT_TEST_SERVER=\"\\\"$AWS_IOT_GAMMA_ENDPOINT\\\"\" -DIOT_TEST_PORT=443 -DIOT_TEST_ROOT_CA=\"\\\"../credentials/AmazonRootCA1.pem\\\"\" -DIOT_TEST_CLIENT_CERT=\"\\\"../credentials/clientCert.pem\\\"\" -DIOT_TEST_PRIVATE_KEY=\"\\\"../credentials/privateKey.pem\\\"\"";
+      else
+        export AWS_IOT_CREDENTIAL_DEFINES="-DIOT_TEST_SERVER=\"\\\"$AWS_IOT_ENDPOINT\\\"\" -DIOT_TEST_PORT=443 -DIOT_TEST_ROOT_CA=\"\\\"../credentials/AmazonRootCA1.pem\\\"\" -DIOT_TEST_CLIENT_CERT=\"\\\"../credentials/clientCert.pem\\\"\" -DIOT_TEST_PRIVATE_KEY=\"\\\"../credentials/privateKey.pem\\\"\"";
+      fi
```

Comment:
can this common code for IOT_TEST_ROOT_CA, DIOT_TEST_CLIENT_CERT and DIOT_TEST_PRIVATE_KEY be moved out of the if?

Reply:
I had tried commonizing it earlier but had run into TravisCI issues. Anyway, these nested ifs are temporary, and will be removed in a separate PR tomorrow

Reply:
ok